### ISSUE TITLE
Undefined browser export

### ISSUE THREAD

**Issue by camille-hdl** [rollup/plugins#256]
- Rollup Plugin Name: plugin-node-resolve
- Rollup Plugin Version: 7.1.1
- Rollup Version: 2.0.5
- Operating System (or Browser): Chrome 80
- Node Version: 13.7

### How Do We Reproduce?

Everything is in this repo : https://github.com/camille-hdl/rollup-node-resolve-bug-yaml

### Expected Behavior

browser export of `yaml` is defined and accessible by
```js
import YAML from "yaml";

YAML.stringify({ my: "input" });
```

### Actual Behavior

```js
import YAML from "yaml";
// YAML is undefined
```

The README in the repo has more details, but here's the gist of it:

```js
// in rollup.config.js
// ...
        resolve({
            mainFields: ["module", "main", "browser"],
        }),
// ....

// in node_modules/yaml/browser/dist/index.js
export default {
  createNode: createNode,
  defaultOptions: defaultOptions,
  Document: Document,
  parse: parse,
  parseAllDocuments: parseAllDocuments,
  parseCST: parseCST,
  parseDocument: parseDocument,
  stringify: stringify
};

// in dist/main.js
var require$$0 = {
  createNode: createNode,
  defaultOptions: defaultOptions,
  Document: Document$2,
  parse: parse$1,
  parseAllDocuments: parseAllDocuments,
  parseCST: parse,
  parseDocument: parseDocument,
  stringify: stringify
};

var browser = require$$0.default; // ⚠️ require$$0.default is undefined !
```

If I remove `"browser"` from the `mainFields` in the plugin, the problem disappears, but other (unrelated?) circular dependency issues arise.

**Comment by eemeli** [comment]
This was caused by an actual circular dependency in `yaml`, which was fixed in its 1.9.0 release. It didn't cause issues when run in Node.js, but apparently it did in Rollup.

**Comment by shellscape** [comment]
Hey folks. This issue hasn't received any traction for more than 90 days, so we're going to close this for housekeeping. If this is still an ongoing issue, please do consider contributing a Pull Request to resolve it. Further discussion is always welcome even with the issue closed. If anything actionable is posted in the comments, we'll consider reopening it.
